src/services/item/repository.ts

```diff
@@ -198,6 +200,38 @@ export class ItemRepository {
       query.andWhere('item.type IN (:...types)', { types });
     }
 
+    const allKeywords = params?.keywords?.filter((s) => s && s.length);
+    if (allKeywords?.length) {
+      const keywordsString = allKeywords.join(' ');
+      query.andWhere((q) => {
+        // search in english by default
+        q.where("item.search_document @@ plainto_tsquery('english', :keywords)", {
```

Comment:
I didn't remember how the search document works, but why do we need to search in english, using simple and using the member's language? Isn't it possible to only use the item's language for the search ? Also, do you know if adding multiple search conditions impact the performance

Reply:
It works with stemmed vectors:
- english: if I search for "dogs", it will stem to "dog" and can search "dog" in items. Search for "chiens" will look for "chiens" as is.
- member lang: (let's say fr) if I search for "chiens", it will stem to "chien" and search for "chien" in items. Looking for "dogs" will look for "dogs" as is
- simple: if I search for "i am" it will look for "i am". However looking for "i am" in english will remove everything because they are stop words.
- raw ILIKE:  if I search for "www" it will look for "www". However looking for "www" in simple will not find "wwwffff" because vectors match on full words only.

So definitely adding lots of cases will impact performance. But as you see I cannot only look for english, or only for member's language. We can later tune if it impacts the performance too much.

I will add that to the specs.

Reply:
Okay I better understand thanks for your explanation ! I'm just not sure to understand why should we keep english and user's lang instead of only using item's lang (english if not set?). In this way if the item is written in English and set in English (in item settings), if the user:
- Search for "dogs", it will be stem to "dog" and search it in the item.
- Search for "chiens" it will search as it is in the item.

I think it's a good idea to keep simple and raw to handle the other cases.

Sorry if I misunderstood something, I think you're right, we can change it later if there is some performance issues.